admin/server/auth/mcp_oauth.go

```diff
@@ -0,0 +1,134 @@
+// as per RFC 8414. This endpoint provides information about the OAuth 2.0 authorization server
+// including supported flows, endpoints, and capabilities.
+func (a *Authenticator) handleOAuthAuthorizationServerMetadata(w http.ResponseWriter, r *http.Request) {
+	metadata := oauth.AuthorizationServerMetadata{
+		Issuer:                a.admin.URLs.External(),
+		AuthorizationEndpoint: a.admin.URLs.OAuthAuthorize(),
+		TokenEndpoint:         a.admin.URLs.OAuthToken(),
+		RegistrationEndpoint:  a.admin.URLs.OAuthRegister(),
+		JWKSURI:               a.admin.URLs.OAuthJWKS(),
+		ScopesSupported: []string{
+			"offline_access", // Refresh token support
+		},
+		ResponseTypesSupported: []string{
+			"code", // Authorization code flow
+		},
+		ResponseModesSupported: []string{
+			"query", // Response parameters in query string
+		},
+		GrantTypesSupported: []string{
+			"authorization_code",                           // Authorization code grant
+			"urn:ietf:params:oauth:grant-type:device_code", // Device code grant
+			"refresh_token",                                // Refresh token grant
+		},
+		TokenEndpointAuthMethodsSupported: []string{
+			"none", // Public clients (PKCE)
+		},
+		CodeChallengeMethodsSupported: []string{
+			"S256", // SHA-256 based PKCE
+		},
+		ServiceDocumentation: "https://docs.rilldata.com",
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(metadata); err != nil {
+		internalServerError(w, fmt.Errorf("failed to encode metadata: %w", err))
+		return
+	}
+}
+
+// handleOAuthRegister handles OAuth 2.0 Dynamic Client Registration as per RFC 7591.
+// This endpoint allows MCP clients like Claude Desktop or ChatGPT Desktop to dynamically
+// register and obtain a client_id for use in OAuth flows.
+func (a *Authenticator) handleOAuthRegister(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		http.Error(w, "expected a POST request", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Parse request body
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		internalServerError(w, fmt.Errorf("failed to read request body: %w", err))
+		return
+	}
+
+	var req oauth.ClientRegistrationRequest
+	if err := json.Unmarshal(body, &req); err != nil {
+		http.Error(w, "invalid JSON in request body", http.StatusBadRequest)
+		return
+	}
+
+	// Use client_name if provided, otherwise use a default name
+	displayName := req.ClientName
+	if displayName == "" {
+		displayName = "MCP Client"
+	}
+
+	// Validate redirect_uris - at least one is required for OAuth flows
+	if len(req.RedirectURIs) == 0 {
+		http.Error(w, "at least one redirect_uri is required", http.StatusBadRequest)
+		return
+	}
+
+	// Create a new auth client in the database
+	client, err := a.admin.DB.InsertAuthClient(r.Context(), displayName)
+	if err != nil {
+		internalServerError(w, fmt.Errorf("failed to create auth client: %w", err))
+		return
+	}
```

Comment:
1. If it already exists, probably that shouldn't be an internal error?
2. Can the endpoint be called multiple times for one client? What if the client metadata like redirect_uri changes?
    - I just looked at the spec, not sure, but sounds like maybe `client_name` is not unique? In that case, maybe `FindAuthClientByDisplayName` is not safe?
3. Should we maybe capture some more fields in the DB in case we need them later? For example, maybe also scope, client_uri, contacts, redirect_uris.